src/cmakeTools.ts

```diff
@@ -1182,8 +1182,9 @@ export class CMakeTools implements api.CMakeToolsAPI {
                 log.debug(localize('cannot.copy.compile.commands', 'Cannot copy {1} because it does not exist at {0}', compdbPath, 'compile_commands.json'));
             }
         }
-
-        if (compdbPaths.length > 0) {
+        if (this.workspaceContext.config.skipLoadingCompileCommands) {
```

Comment:
do we still need to find `compdbPaths` if we are going to set `compilationDatabase` to `null`?
I think we can check for `skipLoadingCompileCommands` earlier, before finding all `compdbPaths`

Reply:
The user may still want the copyCompileCommands setting to work